File: solutions/csharp/space-age/1/SpaceAge.cs
public class SpaceAge
{
    private double earthYears;
    public SpaceAge(int seconds)
    {
        this.earthYears = seconds / 31557600.0;
    }

    public double OnEarth()
    {
        return earthYears;
    }

    public double OnMercury()
    {
        return this.earthYears/0.2408467;
    }

    public double OnVenus()
    {
        return this.earthYears/0.61519726;
    }

    public double OnMars()
    {
        return this.earthYears/1.8808158;
    }

    public double OnJupiter()
    {
        return this.earthYears/11.862615;
    }

    public double OnSaturn()
    {
        return this.earthYears/29.447498;
    }

    public double OnUranus()
    {
        return this.earthYears/84.016846;
    }

    public double OnNeptune()
    {
        return this.earthYears/164.79132;
    }
}

File: solutions/cpp/space-age/1/space_age.cpp
#include "space_age.h"

namespace space_age {

// TODO: add your solution here
    space_age::space_age(long seconds) : 
        total_seconds_(seconds), 
        earth_years_(seconds / 31557600.0) {}
    
    double space_age::seconds() const {
        return total_seconds_;
    }
    double space_age::on_earth() const {
        return earth_years_;
    }
    double space_age::on_mercury() const {
        return earth_years_/0.2408467;
    }
    double space_age::on_venus() const {
        return earth_years_/0.61519726;
    }
    double space_age::on_mars() const {
        return earth_years_/1.8808158;
    }
    double space_age::on_jupiter() const {
        return earth_years_/11.862615;
    }
    double space_age::on_saturn() const {
        return earth_years_/29.447498;
    }
    double space_age::on_uranus() const {
        return earth_years_/84.016846;
    }
    double space_age::on_neptune() const {
        return earth_years_/164.79132;
    }

}  // namespace space_age
